examples/cis2-wccd/src/lib.rs

```diff
@@ -123,6 +154,140 @@ struct UpgradeParams {
     migrate: Option<(OwnedEntrypointName, OwnedParameter)>,
 }
 
+/// The return type for the contract function `view`.
+#[derive(Serialize, SchemaType)]
+struct ReturnBasicState {
+    /// The admin address can upgrade the contract, pause and unpause the
+    /// contract, transfer the admin address to a new address, set
+    /// implementors, and update the metadata URL in the contract.
+    admin:  Address,
+    /// Contract is paused if `paused = true` and unpaused if `paused = false`.
+    paused: bool,
+    /// The metadata URL following the specification RFC1738.
+    #[concordium(size_length = 2)]
+    url:    String,
+}
+
+/// A dummy `WccdEventSchema` struct to implement an event schema including both
+/// (the events specified in this contract and the events specified in the CIS-2
+/// library).
+#[allow(dead_code)]
+struct WccdEventSchema<T: IsTokenId, A: IsTokenAmount> {
+    dummy_t: T,
+    dummy_a: A,
+}
```

Comment:
Why are the two dummy fields needed?

I think the way we want to handle these extra events is to define a type

```
enum WccdEvent {
    CIS2Event(CIS2Event),
   NewAdmin(NewAdminEvent)
}
```

and then define the relevant instances for it.

While this approach you have works, it is not really how we would like to do things.